lua/luaconf.h

```diff
@@ -839,13 +839,28 @@ extern void longjmp(luai_jmpbuf);
 #define LUAI_TRY(L,c,a)		if (setjmp(((c)->b)) == 0) { a }
 
 /* time.h */
-#include <linux/time.h>
 #include <linux/ktime.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,6,0)
+#include <linux/time.h>
+#else
+#ifdef LUA_32BITS
+#error "kernel versions from 5.6.0 onwards do not support 32 bit time values"
+#endif
+#include <linux/time64.h>
+#endif
 static inline int time(void *p)
 {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,6,0)
```

Comment:
please, don't duplicate this logic.. let's just create an abstraction layer to time..